src/main.c

```diff
@@ -0,0 +1,257 @@
+#include <sys/epoll.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+#include <arpa/inet.h>
+
+#include "lib.h"
+
+/*
+ * Write n bytes from buf to the provided fd, retrying short writes until
+ * we finish or hit an error. Assumes fd is blocking and therefore doesn't
+ * handle EAGAIN.
+ */
+void
+write_all(int fd, const char *buf, int n)
+{
+  int m = 0;
+  while(n > 0) {
+    m = write(fd, buf, n);
+    if(m < 0) {
+      perror("writing to stdout");
+      exit(1);
+    }
+    if(m == 0) {
+      fprintf(stderr, "early EOF when writing to stdout\n");
+      exit(1);
+    }
+    n -= m;
+  }
+}
+
+/*
+ * Connect to the given hostname on port 443 and return the file descriptor of
+ * the socket. On error, print the error and return -1. Caller is responsible
+ * for closing socket.
+ */
+int
+make_conn(const char *hostname)
+{
+  struct addrinfo *getaddrinfo_output, *rp;
```

Comment:
nit: many style guides recommend against this kind of declaration. e.g. [CMU](https://wiki.sei.cmu.edu/confluence/display/c/DCL04-C.+Do+not+declare+more+than+one+variable+per+declaration)